src/basic/virt.c

```diff
@@ -472,6 +472,20 @@ int detect_container(void) {
                 goto finish;
         }
 
+        /* Docker creates /.dockerenv or /run/.dockerenv inside the container.
+         * https://github.com/moby/moby/issues/18355 */
+        if (access("/.dockerenv", F_OK) >= 0 || access("/run/.dockerenv", F_OK) >= 0) {
+                r = VIRTUALIZATION_DOCKER;
+                goto finish;
+        }
+
+        /* Podman instead creates /run/.containerenv inside the container.
+         * https://github.com/containers/podman/issues/3586#issuecomment-661918679 */
+        if (access("/run/.containerenv", F_OK) >= 0) {
+                r = VIRTUALIZATION_PODMAN;
+                goto finish;
+        }
```

Comment:
hmm the podman people don't set the $container env var for PID 1? I think they do set that, no? Do we really need to explicitly check for ths file hence if they set the env var? Why have a secondary check for this?

Reply:
They do, but in the linked comment they are saying it's overridden sometimes by distros (RHEL):

> We make a guarantee that the container environment variable will be set inside of a Podman container, and that if not overridden by the container or user it will be set to podman. Red Hat images happen to override by default, which is somewhat annoying.
>
> The best check for container=oci being Docker or Podman would be to look for /run/.containerenv (Podman) vs /run/.dockerenv (Docker)